rclpy/src/rclpy/_rclpy.c

```diff
@@ -713,9 +713,19 @@ rclpy_send_request(PyObject * Py_UNUSED(self), PyObject * args)
   assert(convert_from_py != NULL &&
     "unable to retrieve convert_from_py function, type_support mustn't have been imported");
 
+  PyObject * pydestroy_ros_message = PyObject_GetAttrString(pymetaclass, "_DESTROY_ROS_MESSAGE");
+
+  typedef void * (* destroy_ros_message_signature)(void *);
+  destroy_ros_message_signature destroy_ros_message =
+    (destroy_ros_message_signature)PyCapsule_GetPointer(pydestroy_ros_message, NULL);
+
+  assert(destroy_ros_message != NULL &&
+    "unable to retrieve destroy_ros_message function, type_support mustn't have been imported");
```

Comment:
We'll have to go through all "equivalent" asserts and replace them anyway so I'm not advocating strongly for fixing this in this PR, but I see your point. If you think this is a required change for this PR to get in, please request a change here with an advice on the type of error that should be thrown.